ngraph/test/type_prop/broadcast.cpp

```diff
@@ -706,5 +1075,121 @@ TEST(type_prop, broadcast_v3_output_rank_deduced_from_new_shape_input)
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).rank().is_dynamic());
+
+    // constant target shape
+    const auto target_shape_const = op::Constant::create(element::i64, {3}, {2, 4, 6});
+    broadcast_v3 = make_shared<op::v3::Broadcast>(arg, target_shape_const, "BIDIRECTIONAL");
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).rank().is_static());
+    ASSERT_EQ(broadcast_v3->get_output_partial_shape(0).rank().get_length(), 4);
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).is_dynamic());
+}
+
+TEST(type_prop, broadcast_v3_bidirectional_static_shape_input)
+{
+    const auto arg = make_shared<op::Parameter>(element::f32, PartialShape{1, 2, 3, 1});
+
+    // dynamic target shape
+    auto target_shape = make_shared<op::Parameter>(element::i64, PartialShape::dynamic());
+    auto broadcast_v3 = make_shared<op::v3::Broadcast>(arg, target_shape, "BIDIRECTIONAL");
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).rank().is_dynamic());
+
+    // static rank target shape
+    target_shape = make_shared<op::Parameter>(element::i64, PartialShape::dynamic(1));
+    broadcast_v3 = make_shared<op::v3::Broadcast>(arg, target_shape, "BIDIRECTIONAL");
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).rank().is_dynamic());
+
+    // constant target shape
+    auto target_shape_const = op::Constant::create(element::i64, {4}, {2, 2, 3, 2});
+    broadcast_v3 = make_shared<op::v3::Broadcast>(arg, target_shape_const, "BIDIRECTIONAL");
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).rank().is_static());
+    ASSERT_EQ(broadcast_v3->get_output_partial_shape(0).rank().get_length(), 4);
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).is_static());
+    ASSERT_EQ(broadcast_v3->get_output_partial_shape(0), (PartialShape{2, 2, 3, 2}));
+
+    target_shape_const = op::Constant::create(element::i64, {4}, {5, 2, 3, 7});
+    broadcast_v3 = make_shared<op::v3::Broadcast>(arg, target_shape_const, "BIDIRECTIONAL");
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).rank().is_static());
+    ASSERT_EQ(broadcast_v3->get_output_partial_shape(0).rank().get_length(), 4);
+    ASSERT_TRUE(broadcast_v3->get_output_partial_shape(0).is_static());
+    ASSERT_EQ(broadcast_v3->get_output_partial_shape(0), (PartialShape{5, 2, 3, 7}));
+}
+
+TEST(type_prop, broadcast_v3_bidirectional_partially_dynamic_input)
+{
+    const auto target_shape =
+        op::Constant::create(element::i64, Shape{4}, vector<int64_t>{1, 1, 50, 50});
+
+    auto data = make_shared<op::Parameter>(element::f32, PartialShape{16, 1, Dimension::dynamic()});
+    auto bc = make_shared<op::v3::Broadcast>(data, target_shape, "BIDIRECTIONAL");
+    ASSERT_TRUE(bc->get_output_partial_shape(0).rank().is_static());
+    ASSERT_EQ(bc->get_output_partial_shape(0).rank().get_length(), 4);
+    ASSERT_EQ(bc->get_output_partial_shape(0), (PartialShape{1, 16, 50, Dimension::dynamic()}));
```

Comment:
We could define the shape fully, aren't we? It should be {1, 16, 50, 50} isn't it?

Reply:
Good catch. I think that you are right, but only for scenario when target_dimension != 1 (in such case output dimension can be greater than target dimension).
I relaxed `get_result_shape_bidirectional` method.